src/plugins/intel_gna/src/gna_graph_compiler.cpp

```diff
@@ -663,6 +663,99 @@ void GNAGraphCompiler::finalizeConvolution1DPrimitive(InferenceEngine::CNNLayerP
     }
 }
 
+struct Convolution2DWeightsAdaptorData {
+    size_t original_kernel_width;
+    size_t original_kernel_height;
+    size_t effective_kernel_width;
+    size_t effective_kernel_height;
+    size_t input_channels;
+    size_t convolution_precision;
+    size_t kernels_num;
+    uint8_t* orginal_kernel_data;
+};
+
+class Convolution2DWeightsAdaptor {
+public:
+    Convolution2DWeightsAdaptor(const Convolution2DWeightsAdaptorData& data) : m_data(data) {}
+
+    std::vector<uint8_t> create_weights() {
+        if (is_extendind_weights_needed()) {
+            auto extended_weights = create_extended_weights();
+            return create_transpose_weights(extended_weights.data());
+        }
+
+        return create_transpose_weights(m_data.orginal_kernel_data);
+    }
+
+private:
+    bool is_extendind_weights_needed() {
```

Comment:
is_extending
```suggestion
    bool is_extendind_weights_needed() {
```_weights_needed